### PR TITLE
107-google analytics integration

### PR DESCRIPTION
**Pull request by nasirmids**
Hi Warren, 

Another for review and below is the summary

- Three GA tracking for dev, test and uat
- Initially i tried to put into custom js file as we discussed but there is script tag in the tracking code didnt trigger if i put this in external js file <script async src="https://www.googletagmanager.com/gtag/js?id=UA-132324399-1"></script>
- Denny also had a look but didnt find workable solution.
- Alternatively, There is custom js code where you can call the js file via js code rather via script tag above but that didnt work with our angular like this below
     // var s = document.createElement('script');
    // s.type = 'text/javascript';
    // s.async = true;
    // s.src = 'https://www.googletagmanager.com/gtag/js?id=UA-132324399-1';
    // var x = document.getElementsByTagName('script')[0];
    // x.parentNode.insertBefore(s, x);

- finally, i stop spending much time and i have created 3 index files and in the build i ran a mv command to rename it by environment/branch and it works well. 

The GA dashboard runs off gmail account NMDSdevopsServiceAdm@gmail.com and we can see who is active.

If i find some solution i will do refactoing of that.

Thanks

### CODE REVIEW

**Comment by adamfaryna** [comment]
If we would have something like [dotenv](https://www.npmjs.com/package/dotenv), run variables or env variables we could take leverage of placeholders. The base concept was the best - injecting script element dynamically from the code. 

This solution which moving html around is quite invasive, though.

**Comment by nasirmids** [comment]
> If we would have something like [dotenv](https://www.npmjs.com/package/dotenv), run variables or env variables we could take leverage of placeholders. The base concept was the best - injecting script element dynamically from the code.
> 
> This solution which moving html around is quite invasive, though.

@adamfaryna  correct Adam i have seen DotEnv but not in Anger and can do something like this going forward when i do refactoring at some point.  The first thing would be make the async script work then think about replacement of variable with this dotenv . does it make sense?

File: .travis.yml [hunk 1]
```diff
@@ -47,25 +47,13 @@ script:
...
 before_deploy:    
-    #- if [ "$TRAVIS_BRANCH"=master ]; then B_NAME=master; fi
+    - if [ "$TRAVIS_BRANCH"=master ]; then B_NAME=master; fi
     #- if [ "$TRAVIS_BRANCH"=develop ]; then B_NAME=develop; fi
     #- if [ "$TRAVIS_BRANCH"=test ]; then B_NAME=test; fi
     #- echo $B_NAME
-   #- if [ "$TRAVIS_BRANCH" = master ]; then echo "master"; fi
-   #- if [ "$TRAVIS_BRANCH" = test ]; then echo "test"; fi
-   #- if [ "$TRAVIS_BRANCH" = develop ]; then echo "develop"; fi
+    - if [ "$TRAVIS_BRANCH" = master ]; then echo "master"; mv index_uat.html index.html; fi
```
Comment (wozzer72) [inline .travis.yml]: And whatabout production? Are we expecting an index_prod.html?

Comment (wozzer72) [inline .travis.yml]: I take it we're not using Google Analytics in dev then? What about when we want to demonstrate specific analytic traps going forward?

File: index.html [hunk 2]
```diff
@@ -12,5 +12,12 @@
 </head>
 <body>
   <app-root></app-root>
-  
+  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-132324399-1"></script>
```
Comment (wozzer72) [inline index.html]: The tag here is the only thing changing yes?

File: .travis.yml (hunk 1 above)
Comment (wozzer72) [reply .travis.yml]: Or is dev implied by the default `index.html`???

Comment (nasirmids) [reply .travis.yml]: my fault; the default index file contains dev tracking code

File: index.html (hunk 2 above)
Comment (nasirmids) [reply index.html]: yes thats right; as discussed there are two ways
1- either i replace it as text  from travis build
2- use angular plugin

I opt for 1 one.

File: .travis.yml (hunk 1 above)
Comment (nasirmids) [reply .travis.yml]: well; hopefully this is not needed when i do proper refactoring and we get rid of all files.

**Pull request merged**